src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.iOS.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.IO
+{
+    public static partial class Path
+    {
+        private static string s_defaultTempPath = string.Empty;
+
+        private static string DefaultTempPath
+        {
+            get
+            {
+                s_defaultTempPath = Interop.Sys.SearchPathTempDirectory();
+                if (s_defaultTempPath == null)
+                    throw new System.Runtime.InteropServices.ExternalException();
+                return s_defaultTempPath!;
+            }
+        }
```

Comment:
nit: return the cached variable if it's already set by the previous prop access? we usually don't support / care about changing environment during the lifetime of process.
```suggestion
    private static string DefaultTempPath =>
        s_defaultTempPath ?? (s_defaultTempPath = Interop.Sys.SearchPathTempDirectory()) ??
        throw new System.Runtime.InteropServices.ExternalException();
````